<?xml version="1.0" encoding="UTF-8"?><testrun name="CategoryServiceTests.updateNameCategory" project="MyShopBackend" tests="1" started="1" failures="0" errors="1" ignored="0">
  <testsuite name="com.hcmus.admin.category.CategoryServiceTests" time="0.585" displayname="CategoryServiceTests" uniqueid="[engine:junit-jupiter]/[class:com.hcmus.admin.category.CategoryServiceTests]">
    <testcase name="updateNameCategory" classname="com.hcmus.admin.category.CategoryServiceTests" time="0.585" displayname="updateNameCategory()" uniqueid="[engine:junit-jupiter]/[class:com.hcmus.admin.category.CategoryServiceTests]/[method:updateNameCategory()]">
      <error>org.springframework.dao.DataIntegrityViolationException: could not execute statement [Duplicate entry 'Camera &amp; Photo' for key 'categories.UK_t8o6pivur7nn124jehx7cygw5'] [update categories set alias=?,all_parent_ids=?,enabled=?,image=?,name=?,parent_id=? where id=?]; SQL [update categories set alias=?,all_parent_ids=?,enabled=?,image=?,name=?,parent_id=? where id=?]; constraint [categories.UK_t8o6pivur7nn124jehx7cygw5]&#13;
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:290)&#13;
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:241)&#13;
	at org.springframework.orm.jpa.JpaTransactionManager.doCommit(JpaTransactionManager.java:566)&#13;
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.processCommit(AbstractPlatformTransactionManager.java:794)&#13;
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.commit(AbstractPlatformTransactionManager.java:757)&#13;
	at org.springframework.transaction.interceptor.TransactionAspectSupport.commitTransactionAfterReturning(TransactionAspectSupport.java:676)&#13;
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:426)&#13;
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)&#13;
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)&#13;
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:768)&#13;
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:720)&#13;
	at com.hcmus.admin.category.CategoryService$$SpringCGLIB$$0.save(&lt;generated&gt;)&#13;
	at com.hcmus.admin.category.CategoryServiceTests.updateNameCategory(CategoryServiceTests.java:52)&#13;
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)&#13;
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)&#13;
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)&#13;
Caused by: org.hibernate.exception.ConstraintViolationException: could not execute statement [Duplicate entry 'Camera &amp; Photo' for key 'categories.UK_t8o6pivur7nn124jehx7cygw5'] [update categories set alias=?,all_parent_ids=?,enabled=?,image=?,name=?,parent_id=? where id=?]&#13;
	at org.hibernate.exception.internal.SQLExceptionTypeDelegate.convert(SQLExceptionTypeDelegate.java:62)&#13;
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:58)&#13;
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:108)&#13;
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197)&#13;
	at org.hibernate.engine.jdbc.mutation.internal.AbstractMutationExecutor.performNonBatchedMutation(AbstractMutationExecutor.java:108)&#13;
	at org.hibernate.engine.jdbc.mutation.internal.MutationExecutorSingleNonBatched.performNonBatchedOperations(MutationExecutorSingleNonBatched.java:40)&#13;
	at org.hibernate.engine.jdbc.mutation.internal.AbstractMutationExecutor.execute(AbstractMutationExecutor.java:52)&#13;
	at org.hibernate.persister.entity.mutation.UpdateCoordinatorStandard.doStaticUpdate(UpdateCoordinatorStandard.java:778)&#13;
	at org.hibernate.persister.entity.mutation.UpdateCoordinatorStandard.performUpdate(UpdateCoordinatorStandard.java:324)&#13;
	at org.hibernate.persister.entity.mutation.UpdateCoordinatorStandard.coordinateUpdate(UpdateCoordinatorStandard.java:242)&#13;
	at org.hibernate.persister.entity.AbstractEntityPersister.update(AbstractEntityPersister.java:2810)&#13;
	at org.hibernate.action.internal.EntityUpdateAction.execute(EntityUpdateAction.java:168)&#13;
	at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:632)&#13;
	at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:499)&#13;
	at org.hibernate.event.internal.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:363)&#13;
	at org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:41)&#13;
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:127)&#13;
	at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1403)&#13;
	at org.hibernate.internal.SessionImpl.managedFlush(SessionImpl.java:484)&#13;
	at org.hibernate.internal.SessionImpl.flushBeforeTransactionCompletion(SessionImpl.java:2319)&#13;
	at org.hibernate.internal.SessionImpl.beforeTransactionCompletion(SessionImpl.java:1976)&#13;
	at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.beforeTransactionCompletion(JdbcCoordinatorImpl.java:439)&#13;
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.beforeCompletionCallback(JdbcResourceLocalTransactionCoordinatorImpl.java:169)&#13;
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.commit(JdbcResourceLocalTransactionCoordinatorImpl.java:267)&#13;
	at org.hibernate.engine.transaction.internal.TransactionImpl.commit(TransactionImpl.java:101)&#13;
	at org.springframework.orm.jpa.JpaTransactionManager.doCommit(JpaTransactionManager.java:562)&#13;
	... 13 more&#13;
Caused by: java.sql.SQLIntegrityConstraintViolationException: Duplicate entry 'Camera &amp; Photo' for key 'categories.UK_t8o6pivur7nn124jehx7cygw5'&#13;
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:118)&#13;
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)&#13;
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:912)&#13;
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1054)&#13;
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1003)&#13;
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1312)&#13;
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:988)&#13;
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)&#13;
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)&#13;
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:194)&#13;
	... 35 more&#13;
&#13;
</error>
    </testcase>
  </testsuite>
</testrun>
